src/gui/mzroll/tabledockwidget.cpp

```diff
@@ -1,2532 +1,2847 @@
+          if (group->label == 'b') {
+            selectedGroups.append(group);
+          }
+        } else {
+          selectedGroups.append(group);
         }
+      }
     }
-    return selectedGroups;
+  }
+  return selectedGroups;
 }
 
-//@author:Giridhari -- Refactored this function
-//TODO: To select one or more item in Qtreewidget in peaktable
-PeakGroup* TableDockWidget::getSelectedGroup() { 
-    QTreeWidgetItem *item = treeWidget->currentItem();
-    if (!item) return NULL;
-    QVariant v = item->data(0,Qt::UserRole);
-    PeakGroup*  group =  v.value<PeakGroup*>();
-    if ( group != NULL ) { return group; }
-    else
+PeakGroup *TableDockWidget::getSelectedGroup() {
+  QTreeWidgetItem *item = treeWidget->currentItem();
+  if (!item)
+    return NULL;
+  QVariant v = item->data(0, Qt::UserRole);
+  PeakGroup *group = v.value<PeakGroup *>();
+  if (group != NULL) {
+    return group;
+  } else
     return NULL;
 }
 
 void TableDockWidget::setGroupLabel(char label) {
-    Q_FOREACH(QTreeWidgetItem* item, treeWidget->selectedItems() ) {
-        if (item) {
-            QVariant v = item->data(0,Qt::UserRole);
-            PeakGroup*  group =  v.value<PeakGroup*>();
-            if ( group != NULL ) {
-                 group->setLabel(label);
-            }
-            updateItem(item);
-        }
+  Q_FOREACH (QTreeWidgetItem *item, treeWidget->selectedItems()) {
+    if (item) {
+      QVariant v = item->data(0, Qt::UserRole);
+      PeakGroup *group = v.value<PeakGroup *>();
+      if (group != NULL) {
+        group->setLabel(label);
+      }
+      updateItem(item);
     }
-    updateStatus();
+  }
+  updateStatus();
 }
 
 void TableDockWidget::deleteGroup(PeakGroup *groupX) {
-    if(!groupX) return;
+  if (!groupX)
+    return;
 
-    int pos=-1;
-    for(int i=0; i < allgroups.size(); i++) {
-        if ( &allgroups[i] == groupX ) {
-            pos=i; break;
-        }
+  int pos = -1;
+  for (int i = 0; i < allgroups.size(); i++) {
+    if (&allgroups[i] == groupX) {
+      pos = i;
+      break;
     }
-    if (pos == -1) return;
-
-    //qDebug() << "Delete pos=" << pos;
-    QTreeWidgetItemIterator it(treeWidget);
-    while (*it) {
-        QTreeWidgetItem* item = (*it);
-        if (item->isHidden()) { ++it; continue; }
-        QVariant v = item->data(0,Qt::UserRole);
-        PeakGroup*  group =  v.value<PeakGroup*>();
-        if ( group != NULL and group == groupX) {
-            item->setHidden(true);
-
-            //Deleteing
-            int posTree = treeWidget->indexOfTopLevelItem(item);
-            if (posTree != -1) treeWidget->takeTopLevelItem(posTree);
-
-            /**
-             * delete name of compound associated with this group stored in <sameMzRtGroups> with
-             * given mz and rt
-             */
-            int intMz=group->meanMz*1e5;
-            int intRt=group->meanRt*1e5;
-            QPair<int,int> sameMzRtGroupIndexHash(intMz,intRt);
-  
```

Comment:
@saifulbkhan Does not make sense to have Scatter plot dialog in the scatterplot table. Can you remove it or add a TODO for this?

Reply:
Sure. Missed this.